GDML output fix

mvicenzi opened the pull request:
Closes #52 

Key changes include:
- Each detector class now returns a top-level `G4LogicalVolume` (instead of `G4AssemblyVolume`).  This eliminates nested assemblies in the final geometry tree.
- `SpectrometerMagnetConstruction` class is renamed `FASER2DetectorConstruction`.
- Hardcoded GDML output name is changed to `FPF_FLArE_geo`.

OlivierSalin commented:
Great, looks good ! Good choice to also rename the FASER2 detector construction ! Thanks Matteo
Just an idea but would it be more convenient for user to have the GDML name as an option parser in the futur or at least changing it depending on detector of the FPF are enabled or not ?

OlivierSalin reviewed (approved):
Looks good to me

mvicenzi commented:
> Great, looks good ! Good choice to also rename the FASER2 detector construction ! Thanks Matteo Just an idea but would it be more convenient for user to have the GDML name as an option parser in the futur or at least changing it depending on detector of the FPF are enabled or not ?

Good point: I added another option in the configuration of the geometry to specify the GDML file name. 
I've also updated all the default geometry options in `macro/geometry_options` to set it appropriately.
